Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -392,5 +394,44 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
             if (LogoffChoice.HasValue && LogoffChoice.Value)
                 NativeMethods.Logoff();
         }
+
+        private void checkRunAtLogOn()
+        {
+            if (Shell.IsCairoUserShell.Equals(true))
+            {
+                chkRunAtLogOn.Visibility = Visibility.Hidden;
+            }
+
+            RegistryKey rKey = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run");
+            List<string> rKeyValueNames = rKey.GetValueNames().ToList();
+
+            if (rKeyValueNames.Contains("Cairo"))
```

Comment:
Since the installer sets this to CairoShell, I don't think this check will work unless it is changed to look for "CairoShell".